### PR TITLE
feat(investor): add investor package

### PR DESCRIPTION
**Pull request by cjthompson**
`investor` provides single-sided staking abstraction that can be used to implement various protocols using a common interface.

### CODE REVIEW

**Review by theoboldfrazier — DISMISSED** [review]

File: packages/investor/src/Investor.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type { ChainTypes } from '@shapeshiftoss/types'
+
+import { InvestorOpportunity } from './InvestorOpportunity'
+
+export interface Investor<C extends ChainTypes, TxType = unknown, MetaData = unknown> {
```
Comment (kaladinlight) [inline packages/investor/src/Investor.ts:5]: doc comments describing how this interface works across multiple investor opportunities (some examples may be useful)

File: packages/investor/src/InvestorOpportunity.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    /**
+     * Asset that represents their position
+     */
+    assetId: string
+    /**
+     * The amount of the position asset, not the underlying asset
+     */
+    balance: BigNumber // This is probably a wallet concern not a opportunity concern
+    /**
+     * The ratio of value between the underlying asset and the position asset
```
Comment (theoboldfrazier) [inline packages/investor/src/InvestorOpportunity.ts:32]:
```suggestion
     * The ratio of value between the underlying asset and the position asset in terms of underlying asset per position asset.
```

File: packages/investor/src/InvestorOpportunity.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    /**
+     * Asset used to pay transaction fees
+     */
+    assetId: string
+  }
+  /**
+   * The estimated return on deposited assets
+   *
+   * Amount is an integer value without precision applied
+   *
```
Comment (cjthompson) [inline packages/investor/src/InvestorOpportunity.ts:51]:
```suggestion
```

File: packages/investor/src/InvestorOpportunity.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+   * Sign and broadcast a previously prepared transaction
+   *
+   * @param deps.wallet - HDWallet instance
+   * @param args.tx - Prepared unsigned transaction (from prepareWithdrawal/Deposit)
+   * @param args.feePriority - Specify the user's preferred fee priority (fast/average/slow)
+   */
+  signAndBroadcast: (
+    deps: { wallet: HDWallet },
+    args: { tx: TxType; feePriority?: FeePriority }
+  ) => Promise<string>
```
Comment (cjthompson) [inline packages/investor/src/InvestorOpportunity.ts:99]:
```suggestion
  signAndBroadcast: ({ wallet: HDWallet, tx: TxType; feePriority?: FeePriority }) => Promise<string>
```

File: packages/investor/src/InvestorOpportunity.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    balance: BigNumber // This is probably a wallet concern not a opportunity concern
+    /**
+     * The ratio of value between the underlying asset and the position asset
+     *
+     * Multiply the position asset by this value to calculate the amount of underlying asset
+     * that will be received for a withdrawal
+     *
+     * Amount is an integer value without precision applied
+     */
+    price: BigNumber
```
Comment (cjthompson) [inline packages/investor/src/InvestorOpportunity.ts:39]: Rename the property and reword the description

Comment (theoboldfrazier) [reply packages/investor/src/InvestorOpportunity.ts:39]: `pricePerUnderlying`, `pricePerUnderlyingAsset` or `pricePerShare` is the best I can think of. I think I prefer `pricePerUnderlying` of those three.

**Review by 0xdef1cafe — DISMISSED** [review]
approved conceptually, will restamp once changes are applied

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**